lisp/kbd-mode.el

```diff
@@ -150,32 +150,42 @@ If SHOW-MACROS is nil, don't highlight macros of the form
         (font-lock-add-keywords 'kbd-mode macro-regexp)
       (font-lock-remove-keywords 'kbd-mode macro-regexp))))
 
+;; How many times have it been done?
+(defun backward-symbol ()
+  "Like `forward-symbol' but backwards."
+  (forward-symbol -1))
+
 ;;; Vars
 
-(defvar kbd-mode-syntax-table nil
-  "The basic syntax table for `kbd-mode'.")
-(setq kbd-mode-syntax-table
-  (let ((table (make-syntax-table)))
+(defvar kbd-mode-syntax-table
+  (let ((st (make-syntax-table)))
     ;; Use ;; for regular comments and #| |# for line comments.
-    (modify-syntax-entry ?\; ". 12b" table)
-    (modify-syntax-entry ?\n "> b"   table)
-    (modify-syntax-entry ?\# ". 14"  table)
-    (modify-syntax-entry ?\| ". 23"  table)
+    (modify-syntax-entry ?\; ". 12b" st)
+    (modify-syntax-entry ?\n "> b"   st)
+    (modify-syntax-entry ?\# ". 14"  st)
+    (modify-syntax-entry ?\| ". 23"  st)
 
     ;; We don't need to highlight brackets, as they're only used inside
     ;; layouts.
-    (modify-syntax-entry ?\[ "."     table)
-    (modify-syntax-entry ?\] "."     table)
+    (modify-syntax-entry ?\[ "."     st)
+    (modify-syntax-entry ?\] "."     st)
 
     ;; We highlight the necessary strings ourselves.
-    (modify-syntax-entry ?\" "."     table)
-    table))
-
-(defvar kbd-mode--font-lock-keywords nil
-  "Keywords to be syntax highlighted.")
-(setq
- kbd-mode--font-lock-keywords
- (let ((kexpr-regexp            (regexp-opt kbd-mode-kexpr            'words))
+    (modify-syntax-entry ?\" "."     st)
+    st)
+  "The basic syntax table for `kbd-mode'.")
+
+(defvar kbd-mode-map
+  (let ((map (make-sparse-keymap)))
+    ;; We want to M-{f/b} on any layout entry, not just on alphabet ;)
```

Comment:
Do we?  To be honest, this seems like more of a personal configuration to me, rather than something that `kbd-mode` should enforse.  I for one quite like that e.g. `forward-word` stops on dashes and something like `tap-hold-next-release` is not treated as a single thing.  

Why do you think this is essential to have?

Reply:
Assume qwerty layout in deflayer
I press 'M-f' to jump from 'k' to 'l' -- working
I press 'M-f' to jump from 'l' to ';' -- not working, I jump to 'ret' instead
Or jumping between '_' symbols
Reason to  jump to symbol is  to change/remove it by pressing C/M-d .
Perhaps, I'd better use 'avy'  or C-s and just type that character ;)
But that way also has it's place, we're not enforcing anything.

> I for one quite like that e.g. forward-word stops on dashes and something like tap-hold-next-release is not treated as a single thing.
Agree. That issue is worse than what current pr solves. Hitting C-f 4-5 times is less costly than 5-10

Reply:
I heard of solution of creating and deleting temporary syntax-trees.
We want to treat symbols as words (if we use forward-word command) only under defalias, deflayer, defsrc, etc

Reply:
I feel like this adds a lot more complexity than what is really necessary for such a small mode; especially considering the spurious gain.  
You are right that something like `avy` or `ace-jump-mode` (or even `isearch-forward` and its variants) is probably better suited if you want to go to a certain place, instead of hammering `M-f` until you're there :)   

I'd say remove this and we can merge the `defvar` change (see comments above)